crmsh/utils.py

```diff
@@ -2287,4 +2253,104 @@ def interface_choice():
     # should consider interface format like "ethx@xxx"
     interface_list = re.findall(r'(?:[0-9]+:) (.*?)(?=: |@.*?: )', out)
     return [nic for nic in interface_list if nic != "lo"]
+
+
+class InterfaceInfo(object):
+    def __init__(self, ipv6=False, second_hb=False):
+        self.ipv6 = ipv6
+        self.nic_info_dict = {}
+        self.second_hb = second_hb
+
+    def get_interface_info(self):
+        cmd = "ip -{} -o addr show".format(6 if self.ipv6 else 4)
+        rc, out, err = get_stdout_stderr(cmd)
+        if rc != 0:
+            raise ValueError(err)
+
+        # format on each line will like:
+        # 2: enp1s0    inet 192.168.122.241/24 brd 192.168.122.255 scope global enp1s0\       valid_lft forever preferred_lft forever
+        for line in out.splitlines():
+            _, nic, _, ip_with_mask, *_ = line.split()
+            if nic == "lo":
+                continue
+            # one nic might configured multi IP addresses
+            if nic not in self.nic_info_dict:
+                self.nic_info_dict[nic] = []
+            self.nic_info_dict[nic].append(Interface(ip_with_mask))
+
+        if not self.nic_info_dict:
+            raise ValueError("No address configured")
+        if self.second_hb and len(self.nic_info_dict) == 1:
+            raise ValueError("Cannot configure second heartbeat line, since only one address available")
+
+    def get_default_ip_list(self, specify_nic_list=[]):
+        _ip_list = []
+        for nic in specify_nic_list:
+            if nic not in self.all_nic:
+                raise ValueError("Failed to detect IP address for {}".format(nic))
+            _ip_list.append(self.nic_info_dict[nic][0].ip)
+        if self.second_hb and len(specify_nic_list) == 1:
+            for nic in self.all_nic:
+                if nic not in specify_nic_list:
+                    _ip_list.append(self.nic_info_dict[nic][0].ip)
+                    break
+        return _ip_list
+
+    @property
+    def all_nic(self):
```

Comment:
Maybe better `nic_list` to be aligned with the other method names

Reply:
changed